SAM_UI_TOOL/SAM_image_segmentation.py

```diff
@@ -55,25 +59,92 @@ def __init__(self, args, master, images: List[str]) -> None:
+            reset_button: A tkinter ttk.Button object that resets the GUI to its default state.
+            submit_button: A tkinter ttk.Button object that generates a mask for the current image.
+            previous_button: A tkinter ttk.Button object that loads the previous image in the images list.
+            next_button: A tkinter ttk.Button object that loads the next image in the images list.
+            done_button: A tkinter ttk.Button object that signals that the user has finished generating masks for the current image.
+            exit_button: A tkinter ttk.Button object that exits the GUI.
+            image_num: An integer that represents the index of the current image in the images list.
+            image: A PIL Image object that represents the current image.
+            orig_resolution: A tuple that contains the original resolution of the current image.
+            draw: A PIL ImageDraw object that is used to draw on the image object.
+            base_image: A PIL Image object that represents the original unmodified current image.
+            draw_base: A PIL ImageDraw object that is used to draw on the base_image object.
+            scan: A tkinter PhotoImage object that represents the current image and is displayed in the GUI.
+            label: A tkinter Label object that displays the scan object in the GUI.
+            label2: A tkinter Label object that displays the index of the current image in the images list.
+            label3: A tkinter Label object that displays the file name of the current image.
+            points_list: A list of tuples that contains the coordinates of points that the user has clicked on the image object.
+            labels: A list of integers that represents the labels assigned to each point in points_list.
+            maskImg: A PIL Image object that represents the mask generated for the current image.
+            sam_checkpoint: A string that represents the file path to the checkpoint of the segmentation model.
+            out_dir: A string that represents the file path to the directory where the generated masks will be saved.
+            model_type: A string that represents the type of the segmentation model.
+            sam: A segmentation model object that is used to generate masks for images.
+            predictor: A SamPredictor object that is used to predict masks for images.
+
+        Methods:
+            __init__(self, args, master, images: List[str]) -> None: Initializes the SAMForm object with the given arguments.
+
+            left_click(self, event):
+                Event handler for left clicks on the scan image. Adds a point to the points_list and draws a circle on the image.
+
+            right_click(self, event):
+                Event handler for right clicks on the scan image. Removes the most recent point from the points_list and redraws the image.
+
+            reset(self):
+                Resets the GUI to its default state.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Add the fact that the list is reset as well

Reply:
sure